modules/dkan/dkan_datastore/dkan_datastore.module

```diff
@@ -500,7 +500,7 @@ function dkan_datastore_go($id = NULL, $class = NULL) {
     $class = variable_get('dkan_datastore_class', 'DkanDatastore');
   }
 
-  return class_exists($class) ? Datastore::instance($class, $id) :
+  return class_exists($class) ? DkanDatastore::instance($id, $class) :
```

Comment:
Is there a reason we wouldn't keep the earlier params to preserve some consistency?